cmd/service-controller/controller.go

```diff
@@ -313,11 +329,25 @@ func (c *Controller) checkServiceFor(desired *ServiceBindings, actual *corev1.Se
 		}
 		if actual.Spec.Ports[0].TargetPort.IntValue() != desired.ingressPort {
 			update = true
+			originalAssignedPort, _ := strconv.Atoi(actual.Annotations[types.OriginalAssignedQualifier])
+			actualTargetPort := actual.Spec.Ports[0].TargetPort.IntValue()
+			if actual.ObjectMeta.Annotations == nil {
```

Comment:
@grs added one extra commit that simply initializes the actual service's annotations map.
This was needed because, when I ran the refactored version of the annotated resources integration test
repeatedly, it was sporadically failing.

Looking at the service-controller's log, I noticed that when it happened the pod was restarted because of a panic caused here, due to annotation map being nil.

After applying this fix, I ran the refactored integration test 100 times in a row and all attempts have passed.